clients/nodejs/validator-registration.js

```diff
@@ -0,0 +1,355 @@
+        this._validator_address.serialize(data);
+
+        // We are done with the first part of the transaction data
+        txns_data.push(data);
+
+        // Extract the voting key
+        const voting_key = Nimiq.BufferUtils.fromHex(this._voting_key.toHex());
+
+        if (voting_key.length != 285) {
+            console.error('Invalid voting key');
+            process.exit(1);
+        }
+
+        // Create the next transactions
+        const vk_unused = new Uint8Array(6);
+
+        for (let i = 0; i < 5; i++) {
+            data = new Nimiq.SerialBuffer(64);
+            type += 1;
+            data.writeUint8(type);
+            data.write(vk_unused);
+            // We read the voting key in 57 bytes chunks
+            data.write(voting_key.read(57));
+            txns_data.push(data);
+        }
+
+        return txns_data;
+    }
+}
+
+function help() {
+    console.log(`Nimiq NodeJS tool to register a new validator
+
+Usage:
+    node validator.js [options]
+
+Description:
+    If no arguments are provided all validator keys are generated and stored in a JSONC file
+    Otherwise, the user must specify the path to the JSONC file with the validator configuration:
+    The Validator Account, the Signing Key and the Voting Key.
+    The nimiq core-web package is necessary to execute this tool, this can be installed via:
+    $npm install @nimiq/core-web@next
+
+Options:
+    --help             Display this help page
+    --validator        Path to the validator specification file (JSONC)
+    --network          The network that should be used to register a validator. 
+                       If this argument is not provided, we connect to the test network by default
+    `);
+}
+
+const START = Date.now();
+const argv = require('minimist')(process.argv.slice(2));
+const Nimiq = require('../../dist/node.js');
+const NimiqPOS = require ('@nimiq/core-web');
+const config = require('./modules/Config.js')(argv);
+const fs = require('fs');
+const JSON5 = require('json5');
+const TXN_VALUE = 1;
+const TXN_FEE = 0;
+
+// This tool uses a nano node
+config.protocol = 'dumb';
+config.type = 'nano';
+config.network = 'test';
+
+Nimiq.Log.instance.level = config.log.level;
+for (const tag in config.log.tags) {
+    Nimiq.Log.instance.setLoggable(tag, config.log.tags[tag]);
+}
+
+for (const key in config.constantOverrides) {
+    Nimiq.ConstantHelper.instance.set(key, config.constantOverrides[key]);
+}
+
+for (const seedPeer of config.seedPeers) {
+    if (!seedPeer.host || !seedPeer.port) {
+        console.error('Seed peers must have host and port attributes set');
+        process.exit(1);
+    }
+}
+
+const TAG = 'Node';
+const $ = {};
+
+(async () => {
+    if (argv.help) {
+        return help();
+    }
+
+    if (!argv.validator){
+        // We are in validator configuration generation mode
+        // We need to generate the validator paramaters and exit the tool
+        console.log('Generating new validator paramaters... \n\n');
```

Comment:
I just ran this script and found that when run again, without the `--validator` argument, it simply overwrites any existing `validator-keys.json` file without warning. I think this is a big problem. It should detect when the file already exists and ask the user if he really wants to overwrite it, aborting if not.